src/main/java/baritone/utils/ToolSet.java

```diff
@@ -132,8 +134,15 @@ public int getBestSlot(Block b, boolean preferSilkTouch, boolean pathingCalculat
             if (Baritone.settings().itemSaver.value && (itemStack.getItemDamage() + Baritone.settings().itemSaverThreshold.value) >= itemStack.getMaxDamage() && itemStack.getMaxDamage() > 1) {
                 continue;
             }
+
             double speed = calculateSpeedVsBlock(itemStack, blockState);
             boolean silkTouch = hasSilkTouch(itemStack);
+
+            if(doesBlockRequireSilkTouch && !silkTouch)
+            {
+                continue;
+            }
```

Comment:
Why don't make it so that there is a higher speed if it does have silk touch, otherwise it won't be able to mine the blocks if you don't have a item with silk touch on it in your inventory.
```suggestion
            if(doesBlockRequireSilkTouch && silkTouch) {
                speed *= 1.2;
            }
```

Also formatting